mycore-user2/src/main/java/org/mycore/user2/MCRUserAttributeMapper.java

```diff
@@ -113,88 +113,106 @@ public boolean mapAttributes(final Object object, final Map<String, ?> attribute
         boolean changed = false;
 
         for (Object annotated : getAnnotated(object)) {
-            MCRUserAttribute attrAnno = null;
+            MCRUserAttribute attrAnno = getMCRUserAttribute(annotated);
+            if (attrAnno == null) {
+                continue;
+            }
+
+            final String name = attrAnno.name().isEmpty() ? getAttriutebName(annotated) : attrAnno.name();
+            final List<Attribute> attribs = attributeMapping.get(name);
 
-            if (annotated instanceof Field field) {
-                attrAnno = field.getAnnotation(MCRUserAttribute.class);
-            } else if (annotated instanceof Method method) {
-                attrAnno = method.getAnnotation(MCRUserAttribute.class);
+            if (attributes == null) {
+                return changed;
```

Comment:
This needs to be continue or the other annotation are skipped.

Reply:
This is correct. Like in the original (line128-191) the rest of the code should be skipped if `attributes==null`

Reply:
The Code is wrapped in a for loop. (line 115 old code, line 114 new code) If attributes == null then you skip all the remaining loops, but the old code continues with the other annotations.

Reply:
On a second look you are right! Thank you. I corrected the problem.